tests/bluetooth/tester/src/gap.c

```diff
@@ -275,21 +275,21 @@ static void start_advertising(const uint8_t *data, uint16_t len)
 			goto fail;
 		}
 
-		ad[adv_len].type = cmd->adv_data[i++];
-		ad[adv_len].data_len = cmd->adv_data[i++];
-		ad[adv_len].data = &cmd->adv_data[i];
+		ad[adv_len].type = cmd->adv_sr_data[i++];
+		ad[adv_len].data_len = cmd->adv_sr_data[i++];
+		ad[adv_len].data = &cmd->adv_sr_data[i];
 		i += ad[adv_len].data_len;
 	}
 
-	for (i = 0, sd_len = 0U; i < cmd->scan_rsp_len; sd_len++) {
+	for (sd_len = 0U; i < cmd->adv_data_len+cmd->scan_rsp_len; sd_len++) {
```

Comment:
I think there are more problems in this function. The ad_len starts at 1, so adv_len should have +1 a couple of places.

`if ((adv_len + 1) >= ARRAY_SIZE(ad)) {` line 273

```
	if (bt_le_adv_start(adv_conn ? BT_LE_ADV_CONN : BT_LE_ADV_NCONN,
			    ad, adv_len + 1, sd_len ? sd : NULL, sd_len) < 0) {
```
line 299-300